File: weeks/02/3_diagonal_difference/diagonal_difference.rb
def diagonalDifference(arr)
    primary_diagonal_sum = 0
    secondary_diagonal_sum = 0
  
    (0...arr.length).each do |i|
      primary_diagonal_sum += arr[i][i]
      secondary_diagonal_sum += arr[i][arr.length - (i + 1)]
    end
  
    return (primary_diagonal_sum - secondary_diagonal_sum).abs
  end
  

File: weeks/02/3_diagonal_difference/diagonal_difference.py
def diagonalDifference(arr):

    primary_diagonal_sum = 0
    secondary_diagonal_sum = 0
    
    for i in range(len( arr )):
        primary_diagonal_sum += arr[ i ][ i ]
        secondary_diagonal_sum += arr[ i ][ len(arr)-(i+1) ]
    
    return abs( primary_diagonal_sum-secondary_diagonal_sum )       
